src/MySqlConnector/MySqlClient/ConnectionPool.cs

```diff
@@ -14,6 +14,13 @@ internal sealed class ConnectionPool
 		{
 			cancellationToken.ThrowIfCancellationRequested();
 
+			// if all sessions are used, see if any have been leaked and can be recovered
+			if (m_sessionSemaphore.CurrentCount == 0)
+			{
+				GC.Collect();
+				RecoverLeakedSessions();
```

Comment:
Pushed an update that tries to throttle it to once per second. The check doesn't use a mutex, so if multiple threads arrive at an empty pool at the same time, they could still all attempt to take the `lock (m_leasedSessions)` lock and block on it. But it should stop *every* thread that arrives at an empty pool from calling `GC.Collect` and taking the lock (as was being done in the initial PR).